### PR TITLE
Server bundler

### PR DESCRIPTION
**Pull request by gazuntype**
this is a big one!

changes are implementation of assetbundle to use our cloud assetbundler. when this is merged, when a user chooses to generate an assetbundle, it creates a unitypackage of their work, uploads it to our server where it gets bundled.

### CODE REVIEW

File: Editor/DevPanel.cs [hunk 1]
```diff
@@ -221,49 +323,69 @@ public class DevPanel : EditorWindow {
...
-            assetBundlePath = $"{assetBundleDirectory}/filter";
+            assetBundlePath = $"{assetBundleDirectory}/filter";*/
             statusBar = "Asset bundle generated";
 
 
             statusBar = "Connecting...";
-            Hash128 hash;
-            if (!BuildPipeline.GetHashForAssetBundle(assetBundlePath, out hash)) {
+            byte[] bytes = File.ReadAllBytes(pathToPackage);
+            Hash128 hash = Hash128.Compute(bytes);
```
Comment (chornesays) [inline Editor/DevPanel.cs:383]: I'm worried about this. When Unity grabs an assetbundle from a server- you have to supply the hash. if the hash doesnt match it doesnt download it.

My hunch is.....this won't actually work. The hash has to come from the assetbundler instead.

File: Editor/DevPanel.cs [hunk 2]
```diff
@@ -325,6 +447,8 @@ public class DevPanel : EditorWindow {
                 success = true;
                 try {
                     await GetPrivateCollection();
+                    GetFiltersOnQueue();
+                    GetFiltersOnQueue();
```
Comment (chornesays) [inline Editor/DevPanel.cs:451]: why is this twice?

**Review by chornesays — APPROVED** [review]
Approved with my two comments

File: Editor/DevPanel.cs (hunk 2 above)
Comment (gazuntype) [reply Editor/DevPanel.cs:451]: ah, this is definitely unintentional. maybe i mistakenly cmd+D. will edit.

**Pull request merged**